src/Block/Page/Footer.php

```diff
@@ -98,8 +102,16 @@ public function getPackageJsonData() {
     public function getScandiPWADirectoryPath() {
         $themeDirectoryPath = null;
 
+        $connection = $this->resourceConnection->getConnection();
+        // fetch the id of the currently applied theme
+        $themeid_query = "SELECT value FROM core_config_data WHERE config_id = 37";
+        $themeid = $connection->fetchAll($themeid_query);
+        // use this id to fetch the applied theme itself
+        $applied_theme_query = "SELECT * FROM theme WHERE theme_id = " . $themeid[0]['value'];
+        $applied_theme = $connection->fetchAll($applied_theme_query);
+
```

Comment:
This is exactly the opposite of what we were talking about.
No database queries. Only going through the directory list and checking files.